core/templates/dev/head/forms/formBuilder.js

```diff
@@ -847,66 +861,77 @@ oppia.config(['$provide', function($provide) {
 
 oppia.directive('textAngularRte', [
     '$filter', 'oppiaHtmlEscaper', 'rteHelperService',
-    function($filter, oppiaHtmlEscaper, rteHelperService) {
-  return {
-    restrict: 'E',
-    scope: {
-      htmlContent: '=',
-      uiConfig: '&'
-    },
-    template: (
-      '<div text-angular="" ta-toolbar="<[toolbarOptionsJson]>" ' +
-      '     ta-paste="stripFormatting($html)" ng-model="tempContent">' +
-      '</div>'),
-    controller: ['$scope', function($scope) {
-      $scope.isCustomizationModalOpen = false;
-      var toolbarOptions = [
-        ['bold', 'italics', 'underline'],
-        ['ol', 'ul', 'pre'],
-        []
-      ];
-
-      rteHelperService.getRichTextComponents().forEach(function(componentDefn) {
-        if (!($scope.uiConfig() && $scope.uiConfig().hide_complex_extensions &&
-            componentDefn.isComplex)) {
-          toolbarOptions[2].push(componentDefn.name);
-        }
-      });
-      $scope.toolbarOptionsJson = JSON.stringify(toolbarOptions);
+    'explorationContextService', 'PAGE_CONTEXT',
+    function(
+      $filter, oppiaHtmlEscaper, rteHelperService,
+      explorationContextService, PAGE_CONTEXT) {
+      return {
+        restrict: 'E',
+        scope: {
+          htmlContent: '=',
+          uiConfig: '&'
+        },
+        template: (
+          '<div text-angular="" ta-toolbar="<[toolbarOptionsJson]>" ' +
+          '     ta-paste="stripFormatting($html)" ng-model="tempContent">' +
+          '</div>'),
+        controller: ['$scope', function($scope) {
+          // Currently, operations affecting the filesystem are allowed only in
+          // the editor context.
+          $scope.canUseFs = (
+            explorationContextService.getPageContext() === PAGE_CONTEXT.EDITOR);
```

Comment:
I thought I had a good reason for using $parent.$parent instead, but just using explorationContextService from the provider seems to work (and I can't remember what the reason was, I was probably just confused).